File: src/main/java/org/compiere/orm/X_AD_Ref_Table.java
package org.compiere.orm;

import kotliquery.Row;
import org.compiere.model.I_AD_Ref_Table;

/**
 * Generated Model for AD_Ref_Table
 *
 * @author iDempiere (generated)
 * @version Release 5.1 - $Id$
 */
public class X_AD_Ref_Table extends PO implements I_AD_Ref_Table {

    /**
     *
     */
    private static final long serialVersionUID = 20171031L;

    /**
     * Standard Constructor
     */
    public X_AD_Ref_Table(int AD_Ref_Table_ID) {
        super(AD_Ref_Table_ID);
        /**
         * if (AD_Ref_Table_ID == 0) { setAD_Display (0); setAD_Key (0); setReferenceId (0);
         * setColumnTableId (0); setEntityType (null); // @SQL=select
         * get_sysconfig('DEFAULT_ENTITYTYPE','U',0,0) from dual setIsValueDisplayed (false); }
         */
    }

    /**
     * Load Constructor
     */
    public X_AD_Ref_Table(Row row) {
        super(row);
    }

    /**
     * AccessLevel
     *
     * @return 4 - System
     */
    protected int getAccessLevel() {
        return accessLevel.intValue();
    }

    public String toString() {
        StringBuffer sb = new StringBuffer("X_AD_Ref_Table[").append(getId()).append("]");
        return sb.toString();
    }

    public org.compiere.model.I_AD_Table getTable() throws RuntimeException {
        return (org.compiere.model.I_AD_Table)
                MTable.get(org.compiere.model.I_AD_Table.Table_Name)
                        .getPO(getRefTableId());
    }

    /**
     * Get Table.
     *
     * @return Database Table information
     */
    public int getRefTableId() {
        Integer ii = (Integer) getValue(COLUMNNAME_AD_Table_ID);
        if (ii == null) return 0;
        return ii;
    }

    /**
     * Set Entity Type.
     *
     * @param EntityType Dictionary Entity Type; Determines ownership and synchronization
     */
    public void setEntityType(String EntityType) {

        setValue(COLUMNNAME_EntityType, EntityType);
    }

    /**
     * Set Display Value.
     *
     * @param IsValueDisplayed Displays Value column with the Display column
     */
    public void setIsValueDisplayed(boolean IsValueDisplayed) {
        setValue(COLUMNNAME_IsValueDisplayed, Boolean.valueOf(IsValueDisplayed));
    }

    @Override
    public int getTableId() {
        return Table_ID;
    }
}

File: src/main/kotlin/org/compiere/orm/X_AD_Ref_Table.kt
package org.compiere.orm

import kotliquery.Row
import org.compiere.model.ReferenceTable
import org.compiere.model.Table
import org.idempiere.common.util.AdempiereSystemError
import software.hsharp.core.orm.*

/**
 * Generated Model for AD_Ref_Table
 *
 * @author iDempiere (generated)
 * @version Release 5.1 - $Id$
 */
open class X_AD_Ref_Table : PO, ReferenceTable {

    override val tableId: Int
        get() = ReferenceTable.Table_ID

    /**
     * Standard Constructor
     */
    constructor(AD_Ref_Table_ID: Int) : super(AD_Ref_Table_ID)

    /**
     * Load Constructor
     */
    constructor(row: Row) : super(row)

    /**
     * AccessLevel
     *
     * @return 4 - System
     */
    override fun getAccessLevel(): Int {
        return ReferenceTable.accessLevel.toInt()
    }

    override fun toString(): String {
        return "X_AD_Ref_Table[$id]"
    }

    override fun getTable(): Table {
        return getTable(Table.Table_Name)
            .getPO(refTableId) ?: throw AdempiereSystemError("Table not found")
    }

    /**
     * Get Table.
     *
     * @return Database Table information
     */
    override fun getRefTableId(): Int {
        return getValue(ReferenceTable.COLUMNNAME_AD_Table_ID) as Int? ?: return 0
    }

    /**
     * Set Entity Type.
     *
     * @param EntityType Dictionary Entity Type; Determines ownership and synchronization
     */
    override fun setEntityType(EntityType: String) {

        setValue(ReferenceTable.COLUMNNAME_EntityType, EntityType)
    }

    /**
     * Set Display Value.
     *
     * @param IsValueDisplayed Displays Value column with the Display column
     */
    override fun setIsValueDisplayed(IsValueDisplayed: Boolean) {
        setValue(ReferenceTable.COLUMNNAME_IsValueDisplayed, IsValueDisplayed)
    }

    companion object {
        private const val serialVersionUID = 20171031L
    }
}
